src/components/AddBoxDialogContainer.js

```diff
@@ -43,6 +43,7 @@ class AddBoxDialogContainer extends React.Component {
                 values.product = firestore.doc("products/" + values.product);
                 values.createdAt = firebase.firestore.FieldValue.serverTimestamp();
                 values.createdBy = firestore.doc(profile.ref);
+                values.boxnumber = Math.floor(Math.random() * 1000000);
```

Comment:
Two things:

1. I wonder if this should be called something like `humanID` or `warehousePosition` or similar, to avoid confusion with the quantity of items in the box.
2. Shouldn't this be `Math.floor(Math.random() * 999999)` otherwise there is a non-zero (but very small) chance that you will get 1000000. You're the mathematician though so maybe I am wrong. ;)

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random

"range 0–1 (inclusive of 0, but not 1)"

Actually you are right - ignore point (2)!